### PR TITLE
Add guidelines for using irepts in the code base

### PR DESCRIPTION
**Pull request by thk123**
Following on from discussions - I think we should move to doing this at all times so that changes like #2205 become trivial.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: a2d307d891fd8e57b86910f178ed4bb38f607685).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/79848178
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

File: CODING_STANDARD.md [hunk 1]
```diff
@@ -194,6 +194,12 @@ Formatting is enforced using clang-format. For more information about this, see
   which safely manages the pointer. As such, `new` should only be used in
   constructors, and `delete` in destructors. Never use `malloc` or `free`.
 
+# CProver conventions
+- Avoid if at all possible using irept methods like `get(ID_name)`, instead cast
+  to a derived type (e.g. `class_typet`) and use the wrapper method `get_name`
+- Use `can_cast_type`/`can_cast_exprt` instead of directly checking the `id()`
```
Comment (majakusber) [inline CODING_STANDARD.md]: Typo: `can_cast_expr`

**Review by owen-mc-diffblue — APPROVED** [review]

**Review by majakusber — APPROVED** [review]

File: CODING_STANDARD.md (hunk 1 above)
Comment (johnnonweiler) [inline CODING_STANDARD.md]: Did you mean `can_cast_expr` (without the 't')?

Comment (thk123) [reply CODING_STANDARD.md]: Good spot!

**Comment by thk123** [comment]
This seems to be failing to download source on windows - any ideas why?

**Comment by majakusber** [comment]
Are AWS results binding for us already? I thought it's not yet fully configured.

**Review by smowton — APPROVED** [review]

**Pull request merged**